Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,213 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "MolSupplier.h"
+
+namespace RDKit {
+namespace GeneralMolSupplier {
+struct SupplierOptions {
+  bool takeOwnership = true;
+  bool sanitize = true;
+  bool removeHs = true;
+  bool strictParsing = true;
+
+  std::string delimiter = "\t";
+  int smilesColumn = 0;
+  int nameColumn = 1;
+  bool titleLine = true;
+
+  std::string nameRecord = "";
+  int confId2D = -1;
+  int confId3D = 0;
+};
+//! current supported file formats
+std::vector<std::string> supportedFileFormats{"sdf", "mae", "maegz", "smi",
+                                              "csv", "txt", "tsv",   "tdt"};
+//! current supported compression formats
+std::vector<std::string> supportedCompressionFormats{"gz"};
+
+//! determines whether file and compression format are valid
+bool validate(std::string& fileFormat, std::string& compressionFormat) {
+  bool flag_fileFormat = true;
+  bool flag_compressionFormat = true;
+
+  //! Case 1: No compression format
+  if (compressionFormat.empty()) {
+    //! Unconventional format types
+    if (fileFormat.compare("maegz") == 0) {
+      fileFormat = "mae";
+      compressionFormat = "gz";
```

Comment:
Is this supposed to change the inputs? I don't expect that from something named "validate".